### PR TITLE
feat: build trade btc

### PR DESCRIPTION
**Pull request by elmutt**
## Description

Support trade out of bitcoin. Necessary for thorchain (behind feature flag)

With this merged, the core thorchain trade logic is complete.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Risk

Low risk. only applies to thorchain which is behind a feature flag

## Testing

Make sure trades still work with thorchain flag disabled

### CODE REVIEW

File: src/components/Trade/hooks/useSwapper/useSwapper.ts [hunk 1]
```diff
@@ -303,9 +303,21 @@ export const useSwapper = () => {
...
+        if (!utxoParams?.bip44Params) throw new Error('no bip44Params')
+        return swapper.buildTrade({
+          chainId: KnownChainIds.BitcoinMainnet,
+          sellAmount: amount,
+          sellAsset,
+          buyAsset,
+          sellAssetAccountNumber: 0,
+          buyAssetAccountNumber: 0,
+          wallet,
+          sendMax: true,
```
Comment (0xdef1cafe) [inline src/components/Trade/hooks/useSwapper/useSwapper.ts:316]: ?

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

File: src/components/Trade/hooks/useSwapper/useSwapper.ts [hunk 2]
```diff
@@ -306,13 +306,25 @@ export const useSwapper = () => {
...
           receiveAddress,
         })
       } else if (sellAsset.chainId === KnownChainIds.BitcoinMainnet) {
-        // TODO do bitcoin specific trade quote including `bip44Params`, `accountType` and `wallet`
-        // They will need to have selected an accountType from a modal if bitcoin
-        throw new Error('bitcoin unsupported')
+        const { accountType, utxoParams } = getBtcUtxoParams(accountSpecifiersList, sellAsset)
+        if (!utxoParams?.bip44Params) throw new Error('no bip44Params')
+        return swapper.buildTrade({
+          chainId: KnownChainIds.BitcoinMainnet,
```
Comment (0xdef1cafe) [inline src/components/Trade/hooks/useSwapper/useSwapper.ts:316]: purely a comment - the swapper should be able to infer the `chainId` from the `sellAsset` making this redundant, but let's not change this now

File: src/components/Trade/hooks/useSwapper/useSwapper.ts [hunk 3]
```diff
@@ -306,13 +306,25 @@ export const useSwapper = () => {
...
+          sellAmount: amount,
+          sellAsset,
+          buyAsset,
+          sellAssetAccountNumber: 0,
+          buyAssetAccountNumber: 0,
```
Comment (0xdef1cafe) [inline src/components/Trade/hooks/useSwapper/useSwapper.ts:321]: similarly these should be captured by `utxoParams` and can be removed in future

**Review by 0xdef1cafe — APPROVED** [review]

File: src/components/Trade/hooks/useSwapper/useSwapper.ts (hunk 2 above)
Comment (kaladinlight) [reply src/components/Trade/hooks/useSwapper/useSwapper.ts:316]: yeah was commenting to @gomesalexandre the same thing on lib side

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**